pkg/pluginmanager_service/plugin_manager.go

```diff
@@ -609,7 +611,7 @@ func (m *PluginManager) initializePlugin(connectionConfigs []*sdkproto.Connectio
 // return whether the plugin manager is shutting down
 func (m *PluginManager) shuttingDown() bool {
 	if !m.shutdownMut.TryLock() {
-		return true
+		return m.shutdown
```

Comment:
I admit this is not perfect as a `TryLock()` failure may happen between `m.shutdownMut.Lock()` and `m.shutdown = true` [above](https://github.com/turbot/steampipe/pull/4342/files#diff-808c81445373b3024f6ce0ffd6fc4f20a75e1aba01d6e18a6951a6324937685bR263).

If that is not desirable, maybe the non-blocking `TryLock()`  usage should be replaced by a blocking `Lock()` altogether.